### DISCUSSION TITLE
Modal child window in fullscreen mode

### DISCUSSION THREAD

**Discussion by CSparn** [alejandroautalan/pygubu-designer#254] [Q&A] [answered]
Hi all,

I am relatively new to pygubu and I am looking for a solution of a problem that seems very simple to me.
I would like to open child windows that are modal (the main window can not be changed while open) but with the possibility to open the window in fullscreen mode. I tried the dialogs but there is no fullscreen possible. A second window with Toplevel can not be opend modal.
Can you help me with this?

Thank you in advance.
Greetings
C:S

**Comment by alejandroautalan** [comment, marked as answer]
> Hi all,
> 
> I am relatively new to pygubu and I am looking for a solution of a problem that seems very simple to me. I would like to open child windows that are modal (the main window can not be changed while open) but with the possibility to open the window in fullscreen mode. I tried the dialogs but there is no fullscreen possible. A second window with Toplevel can not be opend modal. Can you help me with this?
> 
> Thank you in advance. Greetings C:S

Hello @CSparn thanks for trying pygubu.

You can use a Toplevel to create modal windows, but you need to manage the logic to show the window in modal state (search for tkinter examples for modals).

Fortunately, pygubu's Dialog class already does all that. You can design the dialog window in the designer and configure whether it will be modal or not.
I made a minimal example trying to meet the requirements you ask for.
You can see it [here](https://github.com/alejandroautalan/pygubu-designer/tree/master/examples/dialogs/demo5).

Try it and tell me if it's what you're looking for.

Regards
Alejandro A.

**Reply by CSparn** [reply]
Hi Alejandro,

thank you very much for the fast answer and the nice example. Actuallly with pygubu's dialogs I am missing the window behavior (resize, closing on X) a bit. So I think the better solution for me will be a Toplevel modal window like you mentioned. I will have a look at the tkinter modal examples.

Greetings
C.S
